packages/webapp/src/components/Finances/PureFinanceTypeSelection/index.jsx

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import MultiStepPageTitle from '../../PageTitle/MultiStepPageTitle';
+import { IconLink, Main } from '../../Typography';
+import Form from '../../Form';
+import Button from '../../Form/Button';
+import Tiles from '../../Tile/Tiles';
+import { tileTypes } from '../../Tile/constants';
+import { ReactComponent as GearIcon } from '../../../assets/images/gear.svg';
+import styles from './styles.module.scss';
+
+export default function PureFinanceTypeSelection({
+  title,
+  types,
+  leadText,
+  cancelTitle,
+  onContinue,
+  onGoBack,
+  onGoToManageCustomType,
+  isTypeSelected,
+  formatTileData,
+  progressValue,
+  useHookFormPersist,
+  iconLinkId,
+}) {
+  const { t } = useTranslation();
+  const { historyCancel } = useHookFormPersist();
+
+  return (
+    <Form
+      buttonGroup={
+        onContinue && (
+          <Button disabled={!isTypeSelected} onClick={onContinue} type={'submit'} fullLength>
+            {t('common:CONTINUE')}
+          </Button>
+        )
+      }
+    >
+      <MultiStepPageTitle
+        onGoBack={onGoBack}
+        onCancel={historyCancel}
+        cancelModalTitle={cancelTitle}
+        title={title}
+        value={progressValue}
+        style={{ marginBottom: '24px' }}
+      />
+      <Main className={styles.leadText}>{leadText}</Main>
+      <Tiles tileType={tileTypes.ICON_LABEL} tileData={types} formatTileData={formatTileData} />
+      <div className={styles.manageCustomTypeLinkContainer}>
+        <IconLink
+          id={iconLinkId}
+          icon={<GearIcon style={{ transform: 'translate(0px, 2px)' }} />}
+          onClick={onGoToManageCustomType}
+          isIconClickable
+          underlined={false}
+        >
+          {t('SALE.FINANCES.MANAGE_CUSTOM_EXPENSE_TYPE')}
```

Comment:
I think the text is generic - but should the name of the translation be generic too?

Reply:
he text for this translation is :
```
...
"MANAGE_CUSTOM_EXPENSE_TYPE": "Manage custom type",
"OTHER_EXPENSES_LABEL": "Other Expenses",
...
 ```
 
 It looks like you did not add this translation so feel free to ignore. Just thought since the text is already generic. We should rename it "MANAGE_CUSTOM_TYPE"

Reply:
Ah, I see. I am adding real "Manage custom expense types" later, so it would make it less confusing, too.
I don't like that it's under SALE because it will be used for revenue type as well, so I will fix it!